### PR TITLE
Add stopping condition checks for top-k variant identification

### PR DESCRIPTION
**Pull request by amishler**
- Adds a function to check whether a top-k set of variants can be identified for some $$k$$ in a specified range $$k_{\min}, \dots, k_{\max}$$. If so, returns the largest such $$k$$.
- A variant is in the top-k if the lower bound of the latest interval in its confidence sequence is larger than the upper confidence bound of at least `num_variants - k` other variants, with a tolerance $$\varepsilon$$.
- Depends on #5137.

----

> [!IMPORTANT]
> Add `topk.rs` module to identify top-k variants using confidence sequences, with integration and extensive testing.
> 
>   - **Behavior**:
>     - Adds `check_topk_stopping()` in `topk.rs` to identify top-k variants based on confidence sequences.
>     - Introduces `check_topk()` as a wrapper for `check_topk_stopping()` with fixed k.
>     - Handles edge cases like invalid parameters and empty variant sets.
>   - **Integration**:
>     - Integrates with `MeanBettingConfidenceSequence` from `betting_confidence_sequences.rs`.
>     - Adds `topk` module to `lib.rs`.
>   - **Testing**:
>     - Extensive tests in `topk.rs` for various scenarios, including edge cases and epsilon tolerance.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=tensorzero%2Ftensorzero&utm_source=github&utm_medium=referral)<sup> for e1aeddc792036791be8235be7cefbc654c439352. You can [customize](https://app.ellipsis.dev/tensorzero/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: evaluations/src/topk.rs [hunk 1]
```diff
@@ -0,0 +1,630 @@
...
+    upper_bounds.sort_by(|a, b| a.partial_cmp(b).unwrap_or(std::cmp::Ordering::Equal));
+
+    // For each variant, count how many upper bounds its lower bound exceeds
+    // This tells us how many variants it "confidently beats"
+    let mut variants_with_n_beaten: Vec<(&String, usize)> = variant_performance
+        .iter()
+        .map(|(name, cs)| {
+            // Binary search to find how many upper bounds are strictly less than this lower bound
+            let num_beaten = upper_bounds.partition_point(|&ub| ub - epsilon < cs.cs_lower);
+            (name, num_beaten)
```
[the remark this answers is not in the document]
Comment (amishler) [reply evaluations/src/topk.rs]: Good catch - fixed.

**Review by virajmehta — DISMISSED** [review]

**Review by virajmehta — APPROVED** [review]
